ragna/ui/resources/upload.js

```diff
@@ -0,0 +1,51 @@
+
+
+function upload(files, token, informationsEndpoint, final_callback) {
+    var uploaded_documents = [] 
+
+    Array.from(files).map(file => {
+
+        getUploadInformations(file, token, informationsEndpoint).then(
+
+                function(uploadInformations) {
+                        uploadFile(uploadInformations.url, uploadInformations.data, file)
+                        uploaded_documents.push(uploadInformations.document)
+                        if (uploaded_documents.length == files.length) {
+                            final_callback(uploaded_documents);
+                        }
+                }
+        );
+        
+    })
+    
+    
+}
+
+async function getUploadInformations(file, token, informationsEndpoint) {
+        const response = await fetch(
+            `${informationsEndpoint}?name=${file.name}`,
+            {"headers": {"Authorization": `Bearer ${token}`}},
+        );
+        const uploadInformations = await response.json();
+
+        return uploadInformations
+}
+
+async function uploadFile(url, data, file) {
+    var body = new FormData()
+
+    for (const [key, value] of Object.entries(data)) {
+      body.append(key, value);
+    }
```

Comment:
Instead of just pulling out the token, we need to include everything that is returned by the upload info. For our endpoint this is indeed just the token, but for S3 it is a lot more.